app/components/administrative-unit-select.hbs

```diff
@@ -1,7 +1,7 @@
 <div class={{if @error "ember-power-select--error"}} ...attributes>
   <PowerSelect
     @disabled={{if @disabled true false}}
-    @allowClear={{@allowClear}}
+    @allowClear={{not @selectedProvince}}
```

Comment:
I'm not sure we should enforce that we can clear this component only if there is a selected province. It forces the user of the component to use it together with the province component, which might not be relevant for all the use cases. I think that the responsibility to define whether we should allow the clearing is in the hands of the template/controller calling the component instead